crates/hir_def/src/nameres/collector.rs

```diff
@@ -297,7 +301,10 @@ impl DefCollector<'_> {
             self.resolve_imports();
 
             match self.resolve_macros() {
-                ReachedFixedPoint::Yes => break,
+                ReachedFixedPoint::Yes => match self.reseed_with_unresolved_attributes() {
+                    ReachedFixedPoint::Yes => break,
+                    ReachedFixedPoint::No => i += 1,
+                },
```

Comment:
This flipping from Yes to No feels a bit odd -- I'd expect RFP to be monotonic. Nothing too major, but I wonder if we can rewrite this somewhat clearer. At the moment, we have three strategies:

* resolve_imports() has fixed-point loop inside
* resolve_macros() does one iteration always 
* `reseed_with_unresolved_attributes` does one iteration if `resolve_macros` is fixed. 

I wonder if we can write literal three loops here?

```rust
loop {
  loop {
    loop {
        if !self.resolve_some_imports() { break }
    }
    if !self.resolve_some_macros() { break }  
  }
  if !self.resolve_some_attrs() { break }  
}
```

It seems like this would accuratelly capture the structure of the computation here.